### PR TITLE
Implemented a general way to use of prompt widget.

### PR DESCRIPTION
**Pull request by sux2mfgj**
This PR has one bug fix and one additional function.

The bug is when I use NullCompleter at the prompt widget, self.history is not created. 

I implemented a general way to use of prompt widget. A behavior is same as #1090.  I also added a document in a source code, however, I think my English is not good. If you think you need fix document,  fix it, please.

thanks.

### CODE REVIEW

File: libqtile/widget/prompt.py [hunk 1]
```diff
@@ -661,10 +668,56 @@ def cmd_info(self):
...
+            Key([alt, 'shift'], 'a',
+                lazy.widget['prompt'].exec_general(
+                    'section(add)',
+                    'layout',
+                    'add_section'))
+        """
+        try:
+            obj = self.qtile.select([(object_name, selector)])
+        except _SelectError:
+            logger.info("cannot select a object")
```
Comment (flacjacket) [inline libqtile/widget/prompt.py]: I would raise this to `error` log level for all of these errors.

File: libqtile/widget/prompt.py [hunk 2]
```diff
@@ -377,15 +378,21 @@ def __init__(self, name="prompt", **config):
...
+                    # self.history of size does not match.
+                    if len(self.history) != len(self.completers):
+                        self.history = {x: deque(maxlen=self.max_history)
+                                for x in self.completers}
+
                     if self.max_history != \
                        self.history[list(self.history)[0]].maxlen:
                         self.history = {x: deque(self.history[x],
                                                  self.max_history)
-                                        for x in self.completers if x}
```
Comment (flacjacket) [inline libqtile/widget/prompt.py]: Why do you remove all of the `if x` from the dict comprehensions?

File: libqtile/widget/prompt.py (hunk 1 above)
Comment (sux2mfgj) [reply libqtile/widget/prompt.py]: I fixed it by cc676ed2141121dbfaeaeb78907084582fc5686e . thank you for your comment.

File: libqtile/widget/prompt.py (hunk 2 above)
Comment (sux2mfgj) [reply libqtile/widget/prompt.py]: This ```if x``` occurred error at https://github.com/qtile/qtile/blob/develop/libqtile/widget/prompt.py#L442.
Is it better to resolve this problem by another changings?

**Comment by flacjacket** [comment]
Thanks for this, I squashed these diffs and push these changes, thanks!

**Comment by sux2mfgj** [comment]
thank you!

**Pull request closed without merging**